src/tests/suite/daos_obj.c

```diff
@@ -1484,17 +1511,56 @@ punch_simple_internal(void **state, daos_obj_id_t oid)
 	memcpy(rec_fetch, rec_verify, size);
 	assert_memory_equal(rec_fetch, rec_verify, size);
 
-	print_message("\tlookup dkey:%s, akey:'akey0'\n", dkeys[0]);
+	/* Lookup first punched record */
 	lookup_single_with_rxnr(dkeys[0], "akey0", /*idx*/0, rec_fetch,
 				/*iod_size*/0, size, DAOS_TX_NONE, &req);
 	assert_memory_equal(rec_fetch, rec_verify, size);
-	print_message("\trecord punched\n");
+	/* Enumerate first punched record*/
+	memset(&anchor_out, 0, sizeof(anchor_out));
+	enum_num = 2;
+	total_keys = 0;
+	while (!daos_anchor_is_eof(&anchor_out)) {
+		daos_epoch_range_t	eprs[5];
+		daos_recx_t		recxs[5];
 
-	print_message("\tlookup dkey:%s, akey:'akey1'\n", dkeys[0]);
+		enum_num = 2;
```

Comment:
similar to the loop above, I think you can just set it here and not on 1520 and 1533